tests/test_revision_groups.py

```
@@ -0,0 +1,531 @@
+            result = doc.rewrite_paragraph(ref, current.replace("beta", "BETA"))
+
+            # The only physical change happens inside our own pre-session
+            # insertion — no trackable revisions, so no group at all.
+            assert result.group_id is None
+            assert doc._revision_manager._groups == {}
+            assert all(rev.group_id is None for rev in doc.list_revisions())
+            # Placement-agnostic content check: replacement position inside
+            # an own insertion is the pre-existing ISSUES.md #31 ordering
+            # bug, out of scope here.
+            text = _paragraph_text(doc, 1)
+            assert "BETA" in text and "beta" not in text
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)